### PR TITLE
Trend charts — Compare to another column

### LINKED ISSUE #35553: Support comparison with another column
- Support comparison with another column

### PR DESCRIPTION
**Pull request by kulyk**
Resolves #35553

Adds "another column" comparison type for trend charts

### How to verify

1. New > Question > Raw Data > Sample Database > Products
2. Summarize Count, Min of Price by Created At
3. Set visualization to "trend"
4. Visualization settings > Data tab > Comparisons > Value from another column > Pick "Min of Price"
5. Try different values for "Label" and "Column" and ensure the chart calculates and displays the change as expected

### Demo

https://github.com/metabase/metabase/assets/17258145/51dc5320-7c66-4d92-8a18-7d3063c57d8d

### CODE REVIEW

File: frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx [hunk 1]
```diff
@@ -47,13 +56,61 @@ export function SmartScalarComparisonWidget({
...
+          value={
+            selectedValue.type === COMPARISON_TYPES.ANOTHER_COLUMN
+              ? selectedValue
+              : undefined
+          }
+          columns={comparableColumns}
+          onChange={nextValue => {
+            onChange(nextValue);
+            setOpen(false);
+          }}
```
Comment (JesseSDevaney) [inline frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx:79]:
```suggestion
          onChange={nextValue => 
            handleEditedValueChange(nextValue, true);
          }
```

File: frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx [hunk 2]
```diff
@@ -21,14 +21,15 @@ export const getSelectOverrides = (): MantineThemeOverride["components"] => ({
...
-      ...getSelectInputOverrides(theme),
+      ...getSelectInputOverrides(theme, size),
       ...getSelectItemsOverrides(theme, size),
     }),
   },
 });
 
 export const getSelectInputOverrides = (
   theme: MantineTheme,
+  size: MantineSize | number,
```
Comment (JesseSDevaney) [inline frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx:32]: This component is causing some weird styling conflicts for selected and hovered states.

**Initial Open of Popover**
Screenshot from 2023-12-28 17-11-58

**Move mouse outside of main column button**
Screenshot from 2023-12-28 17-12-03

**Hover over Item**
Screenshot from 2023-12-28 17-19-57

**Review by JesseSDevaney — COMMENTED** [review]
Nice work, Anton :heart: 

There are a couple things I think we should address before merging though. I noticed some weird styling issues for the column selector and other nits.

File: frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx [hunk 3]
```diff
@@ -47,13 +56,61 @@ export function SmartScalarComparisonWidget({
...
+            editedValue,
+            selectedValue,
+            optionArgs,
+            onChange: handleEditedValueChange,
+            onChangeTab: setTab,
+          }),
+        )}
+      </Stack>
+    );
+  };
```
[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/SmartScalarSettingsWidgets.tsx]: Mind if we get back to this later? I already have some code in flight for the next milestone, and it'd be annoying to fix merge conflicts.

File: frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx (hunk 2 above)
Comment (kulyk) [reply frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx:32]: Nice catch, although it seems to be a common behavior across all `Select` components ([check out Storybook](https://storybook.metabase.com/?path=/story/inputs-select--default-story) for example). Let's bring that up to the rest of the team

Comment (JesseSDevaney) [reply frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx:32]: Were you going to bring that up to the team, or would you prefer I do it?

**Review by JesseSDevaney — APPROVED** [review]
:rocket:

**Pull request merged**